File: hartshorn-launchpad-samples/basic-hello-world-java/src/main/java/org/dockbox/sample/java/GreetingConfiguration.java
package org.dockbox.sample.java;

import org.dockbox.hartshorn.inject.annotations.PropertyValue;
import org.dockbox.hartshorn.inject.annotations.configuration.Configuration;
import org.dockbox.hartshorn.inject.annotations.configuration.Singleton;
import org.dockbox.hartshorn.launchpad.annotations.LoggerMeta;
import org.slf4j.Logger;

@Configuration
public class GreetingConfiguration {

    @Singleton(lazy = true)
    public GreetingAction greetingAction(
            @PropertyValue(name = "greetings.hello", defaultValue = "Hello there, {}") String helloGreetingTemplate,
            @LoggerMeta(name = "Greeting implementation") Logger logger
    ) {
        return () -> logger.info(helloGreetingTemplate, "World");
    }
}

File: hartshorn-launchpad-samples/basic-hello-world-kotlin/src/main/kotlin/org/dockbox/sample/kotlin/GreetingConfiguration.kt
package org.dockbox.sample.kotlin

import org.dockbox.hartshorn.inject.annotations.PropertyValue
import org.dockbox.hartshorn.inject.annotations.configuration.Configuration
import org.dockbox.hartshorn.inject.annotations.configuration.Singleton
import org.dockbox.hartshorn.launchpad.annotations.LoggerMeta
import org.slf4j.Logger

@Configuration
class GreetingConfiguration {

  @Singleton(lazy = true)
  fun greetingAction(
    @PropertyValue(
      name = "greetings.hello",
      defaultValue = "Hello there, {}"
    ) helloGreetingTemplate: String?,
    @LoggerMeta(name = "Greeting implementation") logger: Logger
  ): GreetingAction {
    return GreetingAction { logger.info(helloGreetingTemplate, "World") }
  }
}
